o4/o4.py

```diff
@@ -1092,13 +1092,43 @@ def move_except(from_dir, to_dir, but_not):
     check_call(cmd)
     if not discard:
         savedir = source.replace('cleaning', time.strftime('cleaned'))  # @%Y-%m-%d,%H:%M'))
+        if os.path.exists(savedir):
+            shutil.rmtree(savedir)
         shutil.move(source, savedir)
-        err_print(f'*** INFO: Directory is clean @{changelist}; detritus is in {savedir}')
+        if os.path.exists(f'{savedir}/.o4'):
+            shutil.rmtree(f'{savedir}/.o4')
+        n_files = rm_empty_dirs(savedir)
+        err_print(f'*** INFO: Directory is clean @{changelist}')
+        err_print(f'*** INFO: {n_files} dirty files remain under {savedir}')
     else:
         assert source.endswith('cleaning')
         shutil.rmtree(source)
 
 
+def rm_empty_dirs(root):
+    '''
+    Given a directory tree, remove all directories that have only
+    directories beneath them. Return the number of non-directory files
+    remaining.
+    '''
```

Comment:
`find . -type -d -exec rmdir -p '{}' ';'` ?

Reply:
Didn't know about -depth.
Man page on mac is quite useful:
  -d same -as -depth
  -depth same as -d
It seems pretty weird to combine that with rmdir -p; I suppose find doesn't get confused.

This works, and returns the number of files, so I'm inclined to leave it unless you have a good reason not to.

Reply:
Leave it. I just wanted to share how I would have done it. Not saying that’s how you would have done it. I was just too lazy to do your real solution. 

File count is great. Might even add a fstat and compute total size?

Reply:
I don't see how that would add value. The person needs to address the files regardless.

Reply:
Yes, with the above error, now they do. Like that a lot.

Reply:
Though of course by that logic, the file count likely adds no value either. :D